wzdx/raw_to_standard/navjoy_568.py

```diff
@@ -0,0 +1,326 @@
+        'current_posted_speed': 'currentPostedSpeed4',
+        'reduced_speed_limit': 'requestedTemporarySpeed4',
+        'start_date': 'workStartDate4',
+        'end_date': 'workEndDate4',
+    }
+]
+
+
+def main():
+    navjoy_file, output_dir = parse_rtdh_arguments()
+    input_file_contents = open(navjoy_file, 'r').read()
+    generated_messages = generate_standard_messages_from_string(
+        input_file_contents)
+
+    generated_files_list = []
+    for message in generated_messages:
+        output_path = f"{output_dir}/standard_568_{message['event']['source']['id']}_{round(message['rtdh_timestamp'])}_{message['event']['detail']['direction']}.json"
+        open(output_path, 'w+').write(json.dumps(message, indent=2))
+        generated_files_list.append(output_path)
+
+    if generated_files_list:
+        print(
+            f"Successfully generated standard message files: {generated_files_list}")
+    else:
+        logging.warning(
+            "Standard message generation failed. See messages printed above")
+
+
+# parse script command line arguments
+def parse_rtdh_arguments():
+    parser = argparse.ArgumentParser(
+        description='Translate NavJoy 568 data to RTDH Standard')
+    parser.add_argument('--version', action='version',
+                        version=f'{PROGRAM_NAME} {PROGRAM_VERSION}')
+    parser.add_argument('navjoyFile', help='navjoy file path')
+    parser.add_argument('--outputDir', required=False,
+                        default='./', help='output directory')
+
+    args = parser.parse_args()
+    return args.navjoyFile, args.outputDir
+
+
+# take in individual message, spit out list of altered unique messages to be translated
+# This function iterates over the list of NUMBERED_KEY_NAMES and correlates them to CORRECT_KEY_NAMES. For each
+# set of numbered key names, generate a copy of the original message, check if the numbered keys exist, if they do
+# then copy those values to the leys in CORRECT_KEY_NAMES. After, if directionOfTraffic yields more than one direction,
+# generate a new message for each direction and save them with the key 'direction'
+
+# TODO: consider deleting all numbered keys after they are copied
+# TODO: consider removing duplicate messages
+def expand_speed_zone(message):
+    try:
+        messages = []
+        for key_set in NUMBERED_KEY_NAMES:
+            if not message.get('data', {}).get(key_set['street_name']):
+                continue
+            new_message = copy.deepcopy(message)
+            for key, value in key_set.items():
+                new_message.get('data', {})[CORRECT_KEY_NAMES[key]] = message.get(
+                    'data', {}).get(value)
+
+            directions = new_message.get('data', {}).get('directionOfTraffic')
+            for direction in get_directions_from_string(directions):
+                new_message_dir = copy.deepcopy(new_message)
+                if new_message_dir.get('data', {}).get('directionOfTraffic'):
```

Comment:
Minor thing here, but if we move this removal of the 'directionOfTraffic' property to outside the for loop, the code will only be ran a single time instead of for each direction. Not huge when only going through a few directions, but if there's lots of messages coming through at a time it could help with performance.